src/App.php

```diff
@@ -1263,24 +1254,33 @@ public function runFrontend()
 			$func($this);
 		}
 
-		if ($this->module_class) {
-			if (! $this->error && $_SERVER['REQUEST_METHOD'] === 'POST') {
-				Core\Hook::callAll($this->module . '_mod_post', $_POST);
-				call_user_func([$this->module_class, 'post']);
-			}
+		if (! $this->error && $_SERVER['REQUEST_METHOD'] === 'POST') {
+			Core\Hook::callAll($this->module . '_mod_post', $_POST);
+			call_user_func([$this->module_class, 'post']);
+		}
 
-			if (! $this->error) {
-				Core\Hook::callAll($this->module . '_mod_afterpost', $placeholder);
-				call_user_func([$this->module_class, 'afterpost']);
-			}
+		if (! $this->error) {
+			Core\Hook::callAll($this->module . '_mod_afterpost', $placeholder);
+			call_user_func([$this->module_class, 'afterpost']);
+		}
 
-			if (! $this->error) {
+		$content = '';
+
+		if (! $this->error) {
+			try {
 				$arr = ['content' => $content];
 				Core\Hook::callAll($this->module . '_mod_content', $arr);
 				$content = $arr['content'];
 				$arr = ['content' => call_user_func([$this->module_class, 'content'])];
 				Core\Hook::callAll($this->module . '_mod_aftercontent', $arr);
 				$content .= $arr['content'];
+			} catch(HTTPException $exception) {
+				header($_SERVER["SERVER_PROTOCOL"] . ' ' . $exception->httpcode . ' ' . $exception->httpdesc);
```

Comment:
Wouldn't it be better if the exception handling had been done in ```src/Network/HTTPException.php``` instead? Since that class is responsible for the handling, I would have thought that everything is covered there.

Reply:
No. The principle of exception-based programming, like events, is that its throwing is independent from its handling. Ideally you throw the exception where the logic flow has to stop immediately, and then you trust the call stack to handle it where appropriate. On the other hand, when you set up a `try`/`catch` block, you know you can handle the described exception here, but you don't know exactly where an exception can be thrown in the call stack delimited by the `try` block.

Reply:
The ```catch``` needs to stay here, okay. But does the code inside this have to stay here as well?

Reply:
I guess it could be outsourced in a specific module, which seems to be the path it's going to take to account for HTTPException default message translation. Was that your question?

Reply:
Yes, it was. While looking at the PR, I had asked myself where the output of the error message is done and looked into ```src/Network/HTTPException.php``` and had been confused that is was mostly empty. Then I searched for the template file name and found it in App.php.

Reply:
Exceptions never hold presentation code, this is why putting a `L10n` call is currently contentious, and for good reasons.